cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetSpecificationUpgrader.java

```diff
@@ -100,9 +100,11 @@ public void upgrade() throws Exception {
                 byte[] colVal = columnEntry.getValue();
                 String specEntry = Bytes.toString(colVal);
                 DatasetSpecification specification = GSON.fromJson(specEntry, DatasetSpecification.class);
-                DatasetSpecification updatedSpec = updateTTLInSpecification(specification, null);
-                colVal = Bytes.toBytes(GSON.toJson(updatedSpec));
-                put.add(familyMap.getKey(), columnMap.getKey(), timeStamp, colVal);
+                if (specification != null) {
```

Comment:
Something else is cheesy here. Can we find out what dataset is causing this, and why it has a null spec? The least we should do here is log a warning about the inconsistency.